Custom preset not saving frame rate setting

### Problem description:

When I create a custom preset using the Quick Sync codec, the saved preset does not retain my FPS setting

### HandBrake version (e.g., 1.3.0):

1.3.3 (2020061300)

### Operating system and version (e.g., Ubuntu 19.10, macOS 10.15 Catalina, Windows 10 1909):

Windows 10 Enterprise v.2004

### Error message text and/or screenshot:

No error messages to speak of. I created a custom QSV encoding preset for HEVC stuff. I set a FPS of 29.97 with peak frame rate turned on. When I reload Handbrake it has reverted the FPS to Same as Source. I change it back and use the "update this preset" function getting feedback that it has updated successfully. When I launch Handbrake later, preset is back to same as source. Obviously this is a somewhat minor, first world problem but if I forget to check it I sometimes have to re-encode. You can see in the log that when I set a peak FPS it is honored, it simply isn't saving that setting for next time. I've tested presets on other codecs, the error seems limited to QSV codec presets.

### HandBrake Activity Log:

An Activity Log is required for support. Please read https://handbrake.fr/docs/en/latest/help/activity-log.html for more information.

~~~
HandBrake 1.3.3 (2020061300)
OS: Microsoft Windows NT 10.0.19041.0
CPU: Intel(R) Core(TM) i7-9700K CPU @ 3.60GHz
Ram: 32438 MB, 
GPU Information:
  Intel(R) UHD Graphics 630 - 27.20.100.9466
  NVIDIA GeForce RTX 2080 - 27.21.14.6589
Screen: 2560x1440
Temp Dir: C:\Users\Derek\AppData\Local\Temp\
Install Dir: C:\Program Files\HandBrake
Data Dir: C:\Users\Derek\AppData\Roaming\HandBrake

-------------------------------------------


# Starting Encode ...

[11:55:47] base preset: Quick Sync 265 (Modified)
[23:55:47] hb_init: starting libhb thread
[23:55:47] Starting work at: Sun Jun 06 23:55:47 2021
[23:55:47] 1 job(s) to process
[23:55:47] json job:
{
  "Audio": {
    "AudioList": [
      {
        "DRC": 0.0,
        "Encoder": "copy:ac3",
        "Gain": 0.0,
        "Mixdown": -1,
        "NormalizeMixLevel": false,
        "Samplerate": 0,
        "Name": "Surround 5.1",
        "Track": 0,
        "DitherMethod": 0
      }
    ],
    "CopyMask": [
      "copy:aac",
      "copy:ac3",
      "copy:dtshd",
      "copy:dts",
      "copy:eac3",
      "copy:flac",
      "copy:mp3",
      "copy:truehd"
    ],
    "FallbackEncoder": "ac3"
  },
  "Destination": {
    "ChapterList": [
      {
        "Name": "Chapter 1"
      }
    ],
    "ChapterMarkers": true,
    "AlignAVStart": true,
    "File": "C:\\Users\\Derek\\Videos\\Zombieland - Double Tap 3.m4v",
    "Mp4Options": {
      "IpodAtom": false,
      "Mp4Optimize": true
    },
    "Mux": "av_mp4"
  },
  "Filters": {
    "FilterList": [
      {
        "ID": 12,
        "Settings": {
          "crop-bottom": "60",
          "crop-left": "0",
          "crop-right": "0",
          "crop-top": "60",
          "height": "360",
          "width": "720"
        }
      },
      {
        "ID": 6,
        "Settings": {
          "mode": "2",
          "rate": "27000000/900900"
        }
      }
    ]
  },
  "PAR": {
    "Num": 32,
    "Den": 27
  },
  "Metadata": {},
  "SequenceID": 0,
  "Source": {
    "Angle": 1,
    "Range": {
      "Type": "chapter",
      "Start": 1,
      "End": 1
    },
    "Title": 1,
    "Path": "C:\\Users\\Derek\\Desktop\\Film Editing\\Zombieland - Double Tap.mkv"
  },
  "Subtitle": {
    "Search": {
      "Burn": false,
      "Default": false,
      "Enable": false,
      "Forced": false
    },
    "SubtitleList": []
  },
  "Video": {
    "Encoder": "qsv_h265_10bit",
    "Level": "4.1",
    "TwoPass": false,
    "Turbo": false,
    "ColorMatrixCode": 0,
    "Options": "b-pyramid=1:ref=4",
    "Preset": "speed",
    "Profile": "main10",
    "Quality": 15.0,
    "QSV": {
      "Decode": true,
      "AsyncDepth": 0
    }
  }
}
[23:55:47] CPU: Intel(R) Core(TM) i7-9700K CPU @ 3.60GHz
[23:55:47]  - Intel microarchitecture Kaby Lake
[23:55:47]  - logical processor count: 8
[23:55:47] Intel Quick Sync Video support: yes
[23:55:47]  - Intel Media SDK hardware: API 1.34 (minimum: 1.3)
[23:55:47]  - H.264 encoder: yes
[23:55:47]     - preferred implementation: hardware (any) via D3D11
[23:55:47]     - capabilities (hardware):  breftype icq+la+i+downs vsinfo opt1 opt2+mbbrc+extbrc+trellis+ib_adapt+nmpslice
[23:55:47]  - H.265 encoder: yes (8bit: yes, 10bit: yes)
[23:55:47]     - preferred implementation: hardware (any) via D3D11
[23:55:47]     - capabilities (hardware):  bpyramid icq vsinfo opt1
[23:55:47] hb_scan: path=C:\Users\Derek\Desktop\Film Editing\Zombieland - Double Tap.mkv, title_index=1
udfread ERROR: ECMA 167 Volume Recognition failed
src/libbluray/disc/disc.c:323: failed opening UDF image C:\Users\Derek\Desktop\Film Editing\Zombieland - Double Tap.mkv
src/libbluray/disc/disc.c:424: error opening file BDMV\index.bdmv
src/libbluray/disc/disc.c:424: error opening file BDMV\BACKUP\index.bdmv
src/libbluray/bluray.c:2585: nav_get_title_list(C:\Users\Derek\Desktop\Film Editing\Zombieland - Double Tap.mkv\) failed
[23:55:47] bd: not a bd - trying as a stream/file instead
libdvdnav: Using dvdnav version 6.0.1
libdvdread: Encrypted DVD support unavailable.
libdvdread:DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.IFO failed
libdvdread:DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.BUP failed
libdvdread: Can't open file VIDEO_TS.IFO.
libdvdnav: vm: failed to read VIDEO_TS.IFO
[23:55:47] dvd: not a dvd - trying as a stream/file instead
Input #0, matroska,webm, from 'C:\Users\Derek\Desktop\Film Editing\Zombieland - Double Tap.mkv':
  Metadata:
    creation_time   : 2020-04-05T22:11:37.000000Z
    encoder         : libmakemkv v1.15.0 (1.3.5/1.4.7) win(x64-release)
    TITLE           : Zombieland: Double Tap
    ARTIST          : Woody Harrelson, Jesse Eisenberg, Emma Stone, Abigail Breslin, Zoey Deutch, Avan Jogia, Rosario Dawson, Luke Wilson, Thomas Middleditch, Victoria Hall, Victor Rivera, Ian Gregg, Devin Mojica, Rachel Luttrell, John Dixon, Ronny Mathew, Jess Durham, Da...
    ACTOR           : Woody Harrelson, Jesse Eisenberg, Emma Stone, Abigail Breslin, Zoey Deutch, Avan Jogia, Rosario Dawson, Luke Wilson, Thomas Middleditch, Victoria Hall, Victor Rivera, Ian Gregg, Devin Mojica, Rachel Luttrell, John Dixon, Ronny Mathew, Jess Durham, Da...
    DIRECTOR        : Ruben Fleischer
    PRODUCER        : Gavin Polone
    SCREENPLAY_BY   : Rhett Reese, Dave Callaham, Paul Wernick
    WRITTEN_BY      : Rhett Reese, Dave Callaham, Paul Wernick
    PRODUCTION_STUDIO: Sony Pictures
    GENRE           : Action, Comedy
    DATE_RELEASED   : 2019-10-18
    DATE_RELEASE    : 2019-10-18
    DATE_RECORDED   : 2019-10-18
    CONTENT_TYPE    : Movie
    DESCRIPTION     : Columbus, Tallahassee, Wichita, and Little Rock move to the American heartland as they face off against evolved zombies, fellow survivors, and the growing pains of the snarky makeshift family.
    SUMMARY         : Columbus, Tallahassee, Wichita, and Little Rock move to the American heartland as they face off against evolved zombies, fellow survivors, and the growing pains of the snarky makeshift family.
    SYNOPSIS        : A decade after Zombieland became a hit film and a cult classic, the lead cast (Woody Harrelson, Jesse Eisenberg, Abigail Breslin, and Emma Stone) have reunited for Zombieland: Double Tap. In the sequel where the comic mayhem stretches from the White House
    LAW_RATING      : R
    RATING          : R
  Duration: 01:39:01.94, start: 0.000000, bitrate: 4752 kb/s
    Chapter #0:0: start 5388.216000, end 5941.936000
    Metadata:
      title           : Chapter 16
    Stream #0:0(eng): Video: mpeg2video (Main), yuv420p(tv, progressive), 720x480 [SAR 32:27 DAR 16:9], SAR 186:157 DAR 279:157, 29.97 fps, 29.97 tbr, 1k tbn, 59.94 tbc
    Metadata:
      BPS-eng         : 4104202
      DURATION-eng    : 01:39:01.936000000
      NUMBER_OF_FRAMES-eng: 142464
      NUMBER_OF_BYTES-eng: 3048363576
      SOURCE_ID-eng   : 0100E0
      _STATISTICS_WRITING_APP-eng: MakeMKV v1.15.0 win(x64-release)
      _STATISTICS_WRITING_DATE_UTC-eng: 2020-04-05 22:11:37
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES SOURCE_ID
    Stream #0:1(eng): Audio: ac3, 48000 Hz, 5.1(side), fltp, 448 kb/s (default)
    Metadata:
      title           : Surround 5.1
      BPS-eng         : 448000
      DURATION-eng    : 01:39:01.920000000
      NUMBER_OF_FRAMES-eng: 185685
      NUMBER_OF_BYTES-eng: 332747520
      SOURCE_ID-eng   : 0180BD
      _STATISTICS_WRITING_APP-eng: MakeMKV v1.15.0 win(x64-release)
      _STATISTICS_WRITING_DATE_UTC-eng: 2020-04-05 22:11:37
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES SOURCE_ID
    Stream #0:2(eng): Audio: ac3, 48000 Hz, stereo, fltp, 192 kb/s
    Metadata:
      title           : Stereo
      BPS-eng         : 192000
      DURATION-eng    : 01:39:01.920000000
      NUMBER_OF_FRAMES-eng: 185685
      NUMBER_OF_BYTES-eng: 142606080
      SOURCE_ID-eng   : 0181BD
      _STATISTICS_WRITING_APP-eng: MakeMKV v1.15.0 win(x64-release)
      _STATISTICS_WRITING_DATE_UTC-eng: 2020-04-05 22:11:37
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES SOURCE_ID
    Stream #0:3: Video: mjpeg (Progressive), yuvj420p(pc, bt470bg/unknown/unknown), 2000x3000, 90k tbr, 90k tbn, 90k tbc (attached pic)
    Metadata:
      filename        : cover.jpg
      mimetype        : image/jpeg
[23:55:47] scan: decoding previews for title 1
[23:55:47] scan: audio 0x1: ac3, rate=48000Hz, bitrate=448000 English (AC3) (5.1 ch) (448 kbps)
[23:55:47] scan: audio 0x2: ac3, rate=48000Hz, bitrate=192000 English (AC3) (2.0 ch) (192 kbps)
[23:55:47] WARNING: Video PAR 32:27 != container PAR 186:157
[23:55:47] scan: 10 previews, 720x480, 23.976 fps, autocrop = 60/60/0/0, aspect 16:9, PAR 32:27
[23:55:47] libhb: scan thread found 1 valid title(s)
[23:55:47] Starting Task: Encoding Pass
[23:55:47] work: only 1 chapter, disabling chapter markers
[23:55:47] job configuration:
[23:55:47]  * source
[23:55:47]    + C:\Users\Derek\Desktop\Film Editing\Zombieland - Double Tap.mkv
[23:55:47]    + title 1, chapter(s) 1 to 1
[23:55:47]    + container: matroska,webm
[23:55:47]    + data rate: 4752 kbps
[23:55:47]  * destination
[23:55:47]    + C:\Users\Derek\Videos\Zombieland - Double Tap 3.m4v
[23:55:47]    + container: MPEG-4 (libavformat)
[23:55:47]      + optimized for HTTP streaming (fast start)
[23:55:47]      + align initial A/V stream timestamps
[23:55:47]  * video track
[23:55:47]    + decoder: mpeg2video
[23:55:47]    + filters
[23:55:47]      + Framerate Shaper (mode=2:rate=27000000/900900)
[23:55:47]        + frame rate: 23.976 fps -> peak rate limited to 29.970 fps
[23:55:47]      + Crop and Scale (width=720:height=360:crop-top=60:crop-bottom=60:crop-left=0:crop-right=0)
[23:55:47]        + source: 720 * 480, crop (60/60/0/0): 720 * 360, scale: 720 * 360
[23:55:47]    + Output geometry
[23:55:47]      + storage dimensions: 720 x 360
[23:55:47]      + pixel aspect ratio: 32 : 27
[23:55:47]      + display dimensions: 853 x 360
[23:55:47]    + encoder: H.265 10-bit (Intel Media SDK)
[23:55:47]      + preset:  speed
[23:55:47]      + options: b-pyramid=1:ref=4
[23:55:47]      + profile: main10
[23:55:47]      + level:   4.1
[23:55:47]      + quality: 15.00 (ICQ)
[23:55:47]      + color profile: 6-1-6
[23:55:47]  * audio track 1
[23:55:47]    + name: Surround 5.1
[23:55:47]    + decoder: English (AC3) (5.1 ch) (448 kbps) (track 1, id 0x1)
[23:55:47]      + bitrate: 448 kbps, samplerate: 48000 Hz
[23:55:47]    + AC3 Passthru
[23:55:47] sync: expecting 142464 video frames
[23:55:47] encqsvInit: using encode-only path
[23:55:47] encqsvInit: H.265/HEVC Main 10 profile @ level 4.1
[23:55:47] encqsvInit: TargetUsage 7 AsyncDepth 5
[23:55:47] encqsvInit: GopRefDist 4 GopPicSize 24 NumRefFrame 4
[23:55:47] encqsvInit: BFramesMax 3 BRefType pyramid
[23:55:47] encqsvInit: RateControlMethod ICQ ICQQuality 15
[23:55:47] sync: first pts video is 0
[23:55:47] sync: "Chapter 1" (1) at frame 1 time 0
[23:55:47] sync: first pts audio 0x1 is 0
[23:55:48] qsv_enc_init: using 'hardware (2) via D3D11' implementation, API: 1.34
[23:58:40] reader: done. 1 scr changes
[23:58:40] work: average encoding speed for job is 825.226746 fps
[23:58:40] vfr: 142464 frames output, 0 dropped and 0 duped for CFR/PFR
[23:58:40] vfr: lost time: 0 (0 frames)
[23:58:40] vfr: gained time: 0 (0 frames) (0 not accounted for)
[23:58:40] ac3-decoder done: 185685 frames, 0 decoder errors
[23:58:40] mpeg2video-decoder done: 142464 frames, 0 decoder errors
[23:58:40] sync: got 142464 frames, 142464 expected
[23:58:40] sync: framerate min 19.980 fps, max 23.976 fps, avg 23.976 fps
[23:58:42] mux: track 0, 142464 frames, 1758164393 bytes, 2367.13 kbps, fifo 2048
[23:58:42] mux: track 1, 185685 frames, 332747520 bytes, 448.00 kbps, fifo 4096
[23:58:42] Finished work at: Sun Jun 06 23:58:42 2021
[23:58:42] libhb: work result = 0

# Encode Completed ...


~~~

--- a/win/CS/HandBrakeWPF/Services/Encode/Model/EncodeTask.cs
+++ b/win/CS/HandBrakeWPF/Services/Encode/Model/EncodeTask.cs
@@ -148,6 +148,7 @@ namespace HandBrakeWPF.Services.Encode.Model
             this.IPod5GSupport = task.IPod5GSupport;
             this.OutputFormat = task.OutputFormat;
             this.OptimizeMP4 = task.OptimizeMP4;
+            this.AlignAVStart = task.AlignAVStart;
 
             /* Other */
             this.MetaData = new MetaData(task.MetaData);

--- a/win/CS/HandBrakeWPF/ViewModels/VideoViewModel.cs
+++ b/win/CS/HandBrakeWPF/ViewModels/VideoViewModel.cs
@@ -988,6 +988,10 @@ namespace HandBrakeWPF.ViewModels
                                          ?? VideoTune.None;
                     }
                 }
+                else if (preset.Task.VideoEncoder == VideoEncoder.VP8 || preset.Task.VideoEncoder == VideoEncoder.VP9)
+                {
+                    this.VideoPresetValue = preset.Task.VideoPreset != null ? this.VideoPresets.IndexOf(preset.Task.VideoPreset) : 0;
+                }
             }
 
             this.ExtraArguments = preset.Task.ExtraAdvancedArguments;
@@ -1515,19 +1519,14 @@ namespace HandBrakeWPF.ViewModels
             this.NotifyOfPropertyChange(() => this.IsTwoPassEnabled);
             this.NotifyOfPropertyChange(() => this.DisplayTwoPass);
 
-            // Handle some quicksync specific options.
-            if (selectedEncoder == VideoEncoder.QuickSync || selectedEncoder == VideoEncoder.QuickSyncH265 || selectedEncoder == VideoEncoder.QuickSyncH26510b)
-            {
-                this.TwoPass = false;
-                this.TurboFirstPass = false;
-                this.SelectedFramerate = null;
-            }
-
             if (selectedEncoder == VideoEncoder.NvencH264 || selectedEncoder == VideoEncoder.NvencH265 
                                                           || selectedEncoder == VideoEncoder.VceH264 
                                                           || selectedEncoder == VideoEncoder.VceH265
                                                           || selectedEncoder == VideoEncoder.MFH264
-                                                          || selectedEncoder == VideoEncoder.MFH265)
+                                                          || selectedEncoder == VideoEncoder.MFH265
+                                                          || selectedEncoder == VideoEncoder.QuickSync
+                                                          || selectedEncoder == VideoEncoder.QuickSyncH265
+                                                          || selectedEncoder == VideoEncoder.QuickSyncH26510b)
             {
                 this.TwoPass = false;
                 this.TurboFirstPass = false;

